helix-core/src/test/java/org/apache/helix/TestListenerCallbackBatchMode.java

```diff
@@ -221,21 +219,15 @@ public void testMixedListener() throws Exception {
   }
 
   private void verifyNonbatchedListeners(final Listener listener) throws Exception {
-    Boolean result = TestHelper.verify(new TestHelper.Verifier() {
-      @Override public boolean verify() {
-        return (listener._instanceConfigChangedCount == _numNode) && (
-            listener._idealStateChangedCount == _numResource);
-      }
-    }, 12000);
+    Boolean result = TestHelper.verify(() -> (listener._instanceConfigChangedCount == _numNode)
+        && (listener._idealStateChangedCount == _numResource), 4000);
```

Comment:
Why change timeout helps? If everything completes earlier, the timeout should not matter.